dhis-2/dhis-api/src/main/java/org/hisp/dhis/predictor/Predictor.java

```diff
@@ -90,6 +90,11 @@
      */
     private Set<OrganisationUnitLevel> organisationUnitLevels;
 
+    /**
+     * Whether data for descendant organisation units should be included
+     */
+    private boolean includeDescendantOrgUnits;
```

Comment:
Instead of using a boolean, could we instead have an enum @jimgrace to make it more future proof? I.e. in event analytics we have an enum `OrganisationUnitSelectionMode`. In this case we could include `SELECTED` (just the org unit itself), `CHILDREN` (at the immediate level below only), `DESCENDANTS` (all org units in sub hierarchy). We could even skip `CHILDREN` in the first version. More could be added later. This is helpful to avoid a breaking database/API change if we in the future get more fine-grained requirements for org unit selection.

Reply:
@larshelge, thanks. Great idea. Done. I also renamed `DataExportParams.includeChildren` to `includeDescendants` because that more accurately says what it does.

Let me know if you like the name of the enum `OrganisationUnitDescendants` or if we should change it. I wanted to make something like `OrganisationUnitSelectionMode` but with fewer choices.

Reply:
Thanks @jimgrace , sounds good. I think "selection" or "selection mode" is more appropriate since `SELECTED` actually implies no descendants at all. Though it is not that important.